rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,327 @@
+        owner_address,
+        storage_id,
+        asset_type as coin_type,
+        is_frozen,
+        amount,
+        "type" as activity_type,
+        is_gas_fee,
+        gas_fee_payer_address,
+        is_transaction_success,
+        entry_function_id_str,
+        block_height,
+        token_standard,
+        transaction_timestamp,
+        inserted_at,
+        storage_refund_amount
+    FROM public.fungible_asset_activities
+    WHERE token_standard = 'v1'
+  ;
+-- CREATE INDEX lm1_ca_ct_a_index ON legacy_migration_v1.coin_activities USING btree (coin_type, amount)
+-- CREATE INDEX lm1_ca_ct_at_a_index ON legacy_migration_v1.coin_activities USING btree (coin_type, activity_type, amount)
+-- CREATE INDEX lm1_ca_oa_ct_at_index ON legacy_migration_v1.coin_activities USING btree (owner_address, coin_type, activity_type, amount)
+-- CREATE INDEX lm1_ca_oa_igf_index ON legacy_migration_v1.coin_activities USING btree (owner_address, is_gas_fee)
+
+-- replace `coin_balances` with `fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.coin_balances AS
+    SELECT
+        transaction_version,
+        owner_address,
+        -- this is mainly for hashing the coin type for primary key
+        substring(asset_type, 66) as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_timestamp,
+        inserted_at
+    FROM public.fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+-- CREATE INDEX lm1_cb_insat_index ON legacy_migration_v1.coin_balances USING btree (inserted_at)
+-- CREATE INDEX lm1_cb_oa_ct_index ON legacy_migration_v1.coin_balances USING btree (owner_address, coin_type)
+-- CREATE INDEX lm1_cb_tv_oa_ct_index ON legacy_migration_v1.coin_balances USING btree (transaction_version, owner_address, coin_type)
+
+-- replace `current_coin_balances` with `current_fungible_asset_balances`
+CREATE OR REPLACE VIEW legacy_migration_v1.current_coin_balances AS
+    SELECT
+        owner_address,
+        -- TODO: we also need an index on this!
+        -- a
+        substring(asset_type, 64) as coin_type_hash,
+        asset_type as coin_type,
+        amount,
+        transaction_version as last_transaction_version,
+        transaction_timestamp as last_transaction_timestamp,
+        inserted_at
+    FROM public.fungible_asset_balances
+    WHERE token_standard = 'v1'
+    ;
+-- CREATE INDEX lm1_ccb_ct_a_index ON legacy_migration_v1.current_coin_balances USING btree (coin_type, amount)
+-- CREATE INDEX lm1_ccb_insat_index ON legacy_migration_v1.current_coin_balances USING btree (inserted_at)
+-- CREATE INDEX lm1_ccb_oa_ct_index ON legacy_migration_v1.current_coin_balances USING btree (owner_address, coin_type)
+
+-- replace `token_activities` with `token_activities_v2`
+-- token_activities_v2.token_data_id is 0x prefixed, but token_activities.token_data_id is not. We need to create an index on the substring
+CREATE OR REPLACE VIEW legacy_migration_v1.token_activities AS
```

Comment:
missing the following columns from token_activities

1. creator_address
2. collection_name
3. name